ikarus/io/resultfunction.hh

```diff
@@ -97,10 +97,10 @@ public:
    * \return String representing the name of the result type
    */
   [[nodiscard]] constexpr std::string name() const override {
-    if constexpr (std::is_same_v<UserFunction, Impl::DefaultUserFunction>)
-      return toString(resType);
-    else
+    if constexpr (requires { UserFunction::name(); })
```

Comment:
Here you require that `UserFunction` has a static method name `UserFunction::name()`, but below you use a normal member function `obj.name()`. You should change this require check using the actual object, or also call the name function below as a static function without object `UserFunction::name()`

Reply:
Ah good point. 
I would only require a normal member function, then the interface is less restrictive. 

Also I made the change because I think a ResultEvaluator doesn't need to have a 'name' function, as it could fall back to the name of the result 

I should also add a Concept for ResultEvaluators